src/plugins/identity/alias.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import {type NodeAddressT} from "../../core/graph";
+import {loginAddress as githubAddress} from "../github/nodes";
+import {userAddress as discourseAddress} from "../discourse/createGraph";
+
+/** An Alias is a string specification of an identity within another plugin.
+ *
+ * For now, the supported alias forms are `github/${githubUsername}` and
+ * `discourse/${discourseUsername}`. As a courtesty, if the user has put an @
+ * before the username, we strip it for them.
+ *
+ * This format has been chosen to be moderately extensible and easy to maintain
+ * by hand, as in the near future the alias files will be maintained by hand.
+ * This system will not scale well when user-provided plugins need to add to the
+ * aliasing scheme, so at that point we will rewrite this system.
+ */
+export type Alias = string;
+
+export function resolveAlias(
+  alias: Alias,
+  discourseUrl: string | null
+): NodeAddressT {
```

Comment:
I believe the error handling may be significantly simplified with a regex.
For example `/^(github|discourse)/@?(\w+)$/`.
If explicit errors about unknown types are desired, removing the explicit options works too.
`/^(\w+)/@?(\w+)$/`

Reply:
Re-wrote it to use a regex per your suggestion.